--- a/grammars/tree-sitter-rust/grammar.js
+++ b/grammars/tree-sitter-rust/grammar.js
@@ -1449,7 +1449,7 @@
         seq('\\', choice(
           /[^xu]/,
           /u[0-9a-fA-F]{4}/,
-          /u{[0-9a-fA-F]+}/,
+          seq('u{', /[0-9a-fA-F]+/, '}'),
           /x[0-9a-fA-F]{2}/,
         )),
         /[^\\']/,
@@ -1462,7 +1462,7 @@
         choice(
           /[^xu]/,
           /u[0-9a-fA-F]{4}/,
-          /u{[0-9a-fA-F]+}/,
+          seq('u{', /[0-9a-fA-F]+/, '}'),
           /x[0-9a-fA-F]{2}/,
         ),
       )),