--- a/grammars/tree-sitter-rust/grammar.js
+++ b/grammars/tree-sitter-rust/grammar.js
@@ -1560,7 +1560,7 @@
         seq('\\', choice(
           /[^xu]/,
           /u[0-9a-fA-F]{4}/,
-          /u\{[0-9a-fA-F]+\}/,
+          seq('u{', /[0-9a-fA-F]+/, '}'),
           /x[0-9a-fA-F]{2}/,
         )),
         /[^\\']/,
@@ -1573,7 +1573,7 @@
         choice(
           /[^xu]/,
           /u[0-9a-fA-F]{4}/,
-          /u\{[0-9a-fA-F]+\}/,
+          seq('u{', /[0-9a-fA-F]+/, '}'),
           /x[0-9a-fA-F]{2}/,
         ),
       )),